proxy/src/coco.rs

```diff
@@ -1,649 +1,358 @@
-}
 
-/// Given a path to a repo returns the list of branches.
-///
-/// # Errors
-///
-/// Will return [`error::Error`] if the repository doesn't exist.
-pub fn local_branches(repo_path: &str) -> Result<Vec<Branch>, error::Error> {
-    let repo = surf::git::Repository::new(repo_path)?;
-    let browser = surf::git::Browser::new(repo)?;
-    let mut branches = browser
-        .list_branches(None)?
-        .into_iter()
-        .map(|b| Branch(b.name.name().to_string()))
-        .collect::<Vec<Branch>>();
-
-    branches.sort();
-
-    Ok(branches)
-}
+    /// Fetch a browser for the `project_urn` we supplied to this function.
+    ///
+    /// TODO(finto): The call to `browser` is not actually selecting the correct browser yet.
+    ///
+    /// # Errors
+    ///
+    /// The function will error if:
+    ///   * A lock was poisioned. See [`Self::with_api`].
+    ///   * The repository could not be created. See [`surf::Repository::new`].
+    pub fn project_repo(&self, _project_urn: &str) -> Result<surf::Repository, error::Error> {
+        // TODO(finto): fetch project meta and build browser
+        let project_name = "git-platinum";
+        let path = self.with_api(|api| api.paths().git_dir().join(project_name))?;
+        // TODO(finto): https://github.com/radicle-dev/radicle-surf/issues/126
+        let repo = surf::Repository::new(path.to_str().expect("failed to get path"))?;
+
+        Ok(repo)
+    }
 
-/// Retrieves the [`Commit`] for the given `sha1`.
-///
-/// # Errors
-///
-/// Will return [`error::Error`] if the project doesn't exist or the surf interaction fails.
-pub fn commit(paths: &Paths, id: &str, sha1: &str) -> Result<Commit, error::Error> {
-    let project_id = project::ProjectId::from_str(id)?;
-    let project = project::Project::open(paths, &project_id)?;
-    let mut browser = match project {
-        project::Project::Git(git_project) => git_project.browser()?,
-    };
+    /// Returns the list of [`librad::project::Project`] known for the configured [`Paths`].
+    ///
+    /// # Errors
+    ///
+    /// The function will error if:
+    ///   * A lock was poisioned. See [`Self::with_api`].
+    pub fn list_projects(&self) -> Result<Vec<project::Project<entity::Draft>>, error::Error> {
+        let projects = self.projects.lock().map_err(|_| error::Error::LibradLock)?;
+        Ok(projects.values().cloned().collect())
+    }
 
-    browser.commit(surf::git::Oid::from_str(sha1)?)?;
+    /// Get the project found at `project_urn`.
+    pub async fn get_project(
+        &self,
+        project_urn: &str,
+    ) -> Result<project::Project<entity::Draft>, error::Error> {
+        // TODO(finto): we need the storage to be a resolver
+        let urn = project_urn.parse()?;
+        let project = self.resolve(&urn).await?;
+        Ok(project)
+    }
 
-    let history = browser.get();
-    let commit = history.first();
+    /// Initialize a [`librad::project::Project`] in the location of the given `path`.
```

Comment:
```suggestion
    /// Initialize a [`librad::project::Project`] in the location of the given `path`.
```

It doesn't take a path anymore.